stdlib/public/TensorFlow/Ops.swift

```diff
@@ -1183,18 +1183,11 @@ public extension Tensor {
     }
     @inline(__always)
     set {
-      let orig = self[index]
-      // FIXME: Currently there's no existing non-in-place slice updating
-      // operator in TensorFlow. But since most scalar types are numeric, we do
-      // a hack by using arithmetic operations. We need to find a proper
-      // solution soon.
-      let diff = Tensor(handle: #tfop("Sub", newValue, orig))
-      let scatteredDiff = Raw.scatterNd(
-        indices: Tensor<Int32>(shape: [1, 1], scalars: [index]),
-        updates: Tensor([diff]),
-        shape: shapeTensor
-      )
-      self = Tensor(handle: #tfop("Add", handle, scatteredDiff))
+      let range1: Range = 0..<index
+      let t1 = self[range1]
+      let range2: Range = index+1..<self.shape[0]
```

Comment:
In tensor-only operations, we prefer to avoid using subscript on `self.shape` which will trigger send/receive for uninteresting reasons. You can use `shapeTensor` instead of `shape` because that's tensor-only.

Here's a grotty but efficient solution:
```swift
index+1..<_TFGetScalarOrDie(shapeTensor[0])
```

Reply:
Also, you don't need explicit `self.` here.